core/jvm/src/main/scala/scalaz/zio/system/package.scala

```diff
@@ -0,0 +1,26 @@
+package scalaz.zio
+
+import java.time.Instant
+
+package object system {
+  type TimestampInMillis = Long
+  type TimestampInNanos = Long
+
+  /** Determine the current system time **/
+  def currentTime: IO[Nothing, Instant] = IO.sync(Instant.now)
+
+  /** Determine the current system time in milliseconds **/
+  def currentTimeMillis: IO[Nothing, TimestampInMillis] = IO.sync(System.currentTimeMillis)
+
+  /** Determine the current system time in nanoseconds **/
+  def currentTimeNanos: IO[Nothing, TimestampInNanos] = IO.sync(System.nanoTime)
```

Comment:
`System.nanoTime` does not return wall-clock time, it may be misleading to name it `currentTime`.